pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -11,31 +11,34 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Dict, List, Union
-
-import torch
+from typing import Any, Dict, List, Optional, Tuple, TYPE_CHECKING, Union
 
 from pytorch_lightning.core.step_result import Result
 from pytorch_lightning.trainer.states import TrainerState
 from pytorch_lightning.trainer.supporters import PredictionCollection
-from pytorch_lightning.utilities.apply_func import apply_to_collection
 from pytorch_lightning.utilities.model_helpers import is_overridden
 from pytorch_lightning.utilities.signature_utils import is_param_in_hook_signature
 from pytorch_lightning.utilities.warnings import WarningCache
 
 
+if TYPE_CHECKING:
+    from torch.utils.data import DataLoader
+
+    from pytorch_lightning import Trainer
+    from pytorch_lightning.accelerators.accelerator import _STEP_OUTPUT_TYPE
```

Comment:
Should we start putting types used throughout the codebase in a types file?

Reply:
Yeah, and then only import if `TYPE_CHECKING`

Reply:
yeah, sounds good!

Reply:
Cool, will do separate PR :smiley: